fix: Freeze 'Series column' on detailed results table

Arun-chaitanya opened the pull request:
## Problem

Freeze 'Series column' on detailed results table (resolves https://github.com/PostHog/posthog/issues/11800)

## Changes
@Twixes I did all the changes recommended by you in this PR. (https://github.com/PostHog/posthog/pull/16758). I actually messed up https://github.com/PostHog/posthog/pull/16758, so I am creating a new PR.

<!-- If there are frontend changes, please include screenshots. -->
<!-- If a reference design was involved, include a link to the relevant Figma frame! -->

👉 *Stay up-to-date with [PostHog coding conventions](https://posthog.com/docs/contribute/coding-conventions) for a smoother review.*

## How did you test this code?

<!-- Briefly describe the steps you took. -->
<!-- Include automated tests if possible, otherwise describe the manual testing routine. -->

Arun-chaitanya committed:
added sticky feature to tables

Arun-chaitanya committed:
comment changes on lemonTable sticky feature

Arun-chaitanya committed:
removing commented code

Arun-chaitanya committed:
removing test code

Twixes committed:
Add stickiness shadow

Twixes committed:
Make series clickable

Twixes reviewed (approved):
Thanks, this is a great addition! I also pushed some styling to indicate scrollability with a shadow on the sticky column:

<img width="905" alt="Screenshot 2023-08-01 at 18 43 21" src="https://github.com/PostHog/posthog/assets/4550621/7857c82b-753c-413b-a6e0-ea5c05c5a230">

This is good to go.

Twixes committed:
Fix checkboxes looking broken on Safari

This is not caused by this PR, but now that the series name edit button is inside a checkbox label, things look weird.

Arun-chaitanya commented:
Pleasure is mine. Learning a lot from you guys.
